src/App.tsx

```diff
@@ -17,18 +74,26 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onFilter={handleFilter}
+                onTitleFilter={handleTitleFilter}
+                onClearFilter={handleClearFilter}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!isLoaded && <Loader />}
+              <TodoList
+                todos={visibleTodos}
+                handleSelectedTodo={handleSelectedTodo}
+                selectedTodo={selectedTodo}
+              />
```

Comment:
this should be conditional - if is loading display loader., if not - display list